src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -286,31 +288,29 @@ private static int runCli(CliOptions options, List<File> filesToProcess)
 
         // create config helper object
         if (options.printAst) {
-            // print AST
-            final File file = filesToProcess.get(0);
-            final String stringAst = AstTreeStringPrinter.printFileAst(file,
-                    JavaParser.Options.WITHOUT_COMMENTS);
-            System.out.print(stringAst);
-        }
-        else if (Objects.nonNull(options.xpath)) {
-            final String branch = XpathUtil.printXpathBranch(options.xpath, filesToProcess.get(0));
-            System.out.print(branch);
+            callTreeStringPrinter(filesToProcess.get(0), AstTreeStringPrinter.class);
         }
         else if (options.printAstWithComments) {
-            final File file = filesToProcess.get(0);
-            final String stringAst = AstTreeStringPrinter.printFileAst(file,
-                    JavaParser.Options.WITH_COMMENTS);
-            System.out.print(stringAst);
+            callTreeStringPrinter(filesToProcess.get(0), AstWithCommentsTreeStringPrinter.class);
         }
         else if (options.printJavadocTree) {
-            final File file = filesToProcess.get(0);
-            final String stringAst = DetailNodeTreeStringPrinter.printFileAst(file);
-            System.out.print(stringAst);
+            callTreeStringPrinter(filesToProcess.get(0), DetailNodeTreeStringPrinter.class);
         }
         else if (options.printTreeWithJavadoc) {
-            final File file = filesToProcess.get(0);
-            final String stringAst = AstTreeStringPrinter.printJavaAndJavadocTree(file);
-            System.out.print(stringAst);
+            callTreeStringPrinter(filesToProcess.get(0), AstAndDetailNodeTreeStringPrinter.class);
+        }
+        else if (Objects.nonNull(options.customTreePrinterClass)) {
+            try {
+                callTreeStringPrinter(filesToProcess.get(0),
+                        Class.forName(options.customTreePrinterClass));
+            }
+            catch (ClassNotFoundException | IOException | CheckstyleException ex) {
```

Comment:
No need to catch `CheckstyleException` here. It just wraps one `CheckstyleException` with another:
```bash
java -jar checkstyle.jar -pc com.puppycrawl.tools.checkstyle.utils.CheckUtil

com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to work with custom tree printer
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:308)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:197)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:132)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to work with custom tree printer
	at com.puppycrawl.tools.checkstyle.Main.callTreeStringPrinter(Main.java:360)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:304)
	... 2 more
```